sim-lib/src/lib.rs

```diff
@@ -435,6 +447,23 @@ enum SimulationOutput {
     SendPaymentFailure(Payment, PaymentResult),
 }
 
+type MutRngType = Arc<StdMutex<Box<dyn RngCore + Send>>>;
+
+#[derive(Clone)]
+struct MutRng(pub MutRngType);
+impl MutRng {
```

Comment:
Don't personally feel like adding a wrapper struct on `MutRngType` adds much value (assuming it's there so we can have an `impl` block + constructor) here over just passing around params of the type, but not blocking merge.

Reply:
It makes it easier to create new `MutRng` types within code and in tests. I think it is much cleaner to have `MutRng::new(T)` that `Arc::new(Mutex::new(Box::new(T)))`